README.md

```diff
@@ -1,16 +1,137 @@
+- Bash shell (tested on Mac, Ubuntu, Windows with WSL2)
+  - Will not work in Cloud Shell unless you have a remote dockerd
+- Azure CLI 2.0.72+ ([download](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest))
+- Docker CLI ([download](https://docs.docker.com/install/))
+- Java 8 above ([download](https://nodejs.org/en/download/)) 
+- JQ ([download](https://stedolan.github.io/jq/download/))
+- Completion of the Key Vault, ACR, Azure Monitor and Cosmos DB setup in the Helium [readme](https://github.com/retaildevcrews/helium)
+- Visual Studio Code (optional) ([download](https://code.visualstudio.com/download))
+
+## Setup
+
+The application requires Key Vault and Cosmos DB to be setup per the Helium [readme](https://github.com/retaildevcrews/helium)
+
+- Fork this repo and clone to your local machine
+  - All instructions assume starting from the root of the repo
+
+## Build the container using Docker
+# The unit tests run as part of the Docker build process. You can also run the unit tests manually.
+# For instructions on building the container with ACR, please see the Helium [readme](https://github.com/retaildevcrews/helium)
+'''bash
+
+# Make sure you are in the root of the repo
+# Build the image - this can take 3-5 minutes
+
+docker build -t helium-dev .
+
+```
+## Run the application locally and clean previous packages and unit tests
+
+- The application requires Key Vault and Cosmos DB to be setup per the Helium [readme](https://github.com/retaildevcrews/helium)
+  - You can run the application locally by using Azure CLI cached credentials
+    - You must run az login before this will work
+
+'''bash
+
+# Make sure you are in the root of the repo
+
+export AUTH_TYPE=CLI
+export KEYVAULT_NAME="YOUR_KEYVAULT_NAME"
+
+This command takes about 3-5 minutes:
+
+mvn clean package 
+
+mvn spring-boot:run
+
+# When the previous command shows 'Netty started on port(s): 8080' 
+# test the application (takes about 10 seconds to start) in a new window
+
+curl http://localhost:8080/healthz
+
+# Stop the application by typing Ctrl-C in the Azure CLI terminal window
+Or :::::
+mvn spring-boot:stop
+
+```
+
+## Run the application as a local container instead
+
+'''bash
+
+# Make sure you are in the root of the repo
+# Run the command below to stop the previous instance and free up port 8080:
+
+mvn spring-boot:stop 
+
+# You can also hit Ctrl-C from the az cli window
+
+# Build the image and run the container using Docker 
+#
+# make sure you are in the root of the repo
+
+# this will use Azure CLI cached credentials
+
+# build the image
+
+docker build -t helium-dev .
+
+# Then run the container.  
+
+# First obtain APPLICATIONINSIGHTS_CONNECTION_STRING using the following link:
+#TEST THIS
+https://docs.microsoft.com/en-us/azure/azure-monitor/app/java-get-started?tabs=maven
+
+export APP_INSIGHTS_CONNECTION_STRING=<KEY_FROM_ABOVE_LINK>
```

Comment:
if we use app insights, the env var should be pulled from an az call. The goal is full copy paste for developer experience

Reply:
I added an option in the readme to run w/appinsights or w/o.  I set the appinsights instrumentation key with this:
export APP_INSIGHTS_CONNECTION_STRING="$(az monitor app-insights component show --app inst2-appinsights --resource-group wsrtf-app-rg --output yaml|grep instrumentationKey|awk '{print $2}')"